src/test/java/org/broadinstitute/consent/http/util/JsonSchemaUtilTest.java

```diff
@@ -1,46 +1,71 @@
+            "species": "species",
+            "piName": "PI Name",
+            "dataSubmitterUserId": 1,
+            "dataCustodianEmail": ["email@abc.com"],
+            "publicVisibility": true,
+            "dataAccessCommitteeId": 1,
+            "consentGroups": [{
+              "consentGroupName": "name",
+              "generalResearchUse": true
+            }]
+          }
+          """;
+
   @Test
-  public void testIsValidDataSubmitterObject_v1_case0() {
+  public void testIsValidDatasetRegistrationObject_v1_case0() {
     String instance = "{}";
     JsonSchemaUtil util = new JsonSchemaUtil();
     boolean valid = util.isValidSchema_v1(instance);
     assertFalse(valid);
   }
 
   @Test
-  public void testIsValidDataSubmitterObject_v1_case1() {
-    String instance = "{ " +
-            "  \"studyType\": \"Observational\", " +
-            "  \"studyName\": \"name\", " +
-            "  \"studyDescription\": \"description\", " +
-            "  \"dataTypes\": [\"types\"], " +
-            "  \"fileTypes\": [{ " +
-            "    \"fileType\": \"Arrays\", " +
-            "    \"functionalEquivalence\": \"equivalence\", " +
-            "    \"numberOfParticipants\": 2 " +
-            "  }], " +
-            "  \"phenotypeIndication\": \"phenotype\", " +
-            "  \"species\": \"species\", " +
-            "  \"piName\": \"PI Name\", " +
-            "  \"dataSubmitterUserId\": 1, " +
-            "  \"dataCustodianEmail\": [\"email@abc.com\"], " +
-            "  \"publicVisibility\": true, " +
-            "  \"dataAccessCommitteeId\": 1, " +
-            "  \"consentGroups\": [{ " +
-            "    \"consentGroupName\": \"name\", " +
-            "    \"generalResearchUse\": true " +
-            "  }] " +
-            "}";
+  public void testIsValidDatasetRegistrationObject_v1_case1() {
     JsonSchemaUtil util = new JsonSchemaUtil();
-    boolean valid = util.isValidSchema_v1(instance);
+    boolean valid = util.isValidSchema_v1(datasetRegistrationInstance);
     assertTrue(valid);
   }
+
+  @Test
+  public void testParseDatasetRegistrationObject_v1() {
+    JsonSchemaUtil util = new JsonSchemaUtil();
+    DatasetRegistrationSchemaV1 instance = util.deserializeDatasetRegistration(datasetRegistrationInstance);
+    assertNotNull(instance);
+    assertNotNull(instance.getStudyType());
+    assertNotNull(instance.getStudyName());
+    assertNotNull(instance.getStudyDescription());
+    assertFalse(instance.getDataTypes().isEmpty());
+    assertFalse(instance.getFileTypes().isEmpty());
+    assertNotNull(instance.getPhenotypeIndication());
+    assertNotNull(instance.getSpecies());
+    assertNotNull(instance.getPiName());
+    assertNotNull(instance.getDataSubmitterUserId());
+    assertFalse(instance.getDataCustodianEmail().isEmpty());
+    assertNotNull(instance.getPublicVisibility());
+    assertNotNull(instance.getDataAccessCommitteeId());
+    assertFalse(instance.getConsentGroups().isEmpty());
```

Comment:
should we have more comprehensive testing of e.g., enum parsing / other custom types?

Reply:
We do need more testing - am hoping to do more in a future PR. I anticipate some changes as we integrate the UI and the API, so am holding off on that for now.